src/main/java/com/bravo/user/service/PaymentService.java

```diff
@@ -0,0 +1,37 @@
+package com.bravo.user.service;
+
+import com.bravo.user.dao.model.Payment;
+import com.bravo.user.dao.model.mapper.ResourceMapper;
+import com.bravo.user.dao.repository.PaymentRepository;
+import com.bravo.user.dao.repository.UserRepository;
+import com.bravo.user.exception.DataNotFoundException;
+import com.bravo.user.model.dto.PaymentDto;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@Slf4j
+public class PaymentService {
+
+    private final PaymentRepository paymentRepository;
+    private final UserRepository userRepository;
+    private final ResourceMapper mapper;
+
+    public PaymentService(PaymentRepository paymentRepository, UserRepository userRepository, ResourceMapper mapper) {
+        this.paymentRepository = paymentRepository;
+        this.userRepository = userRepository;
+        this.mapper = mapper;
+    }
+
+    public List<PaymentDto> retrieveByUserId(String userId) {
+        log.info("Retrieving all payments by user Id");
+
+        userRepository.findById(userId).orElseThrow(() -> new DataNotFoundException("User Id does not exist. Please validate input"));
```

Comment:
Does it make sense to declare a `existsById` query instead of doing a search to throw away the result. `exists` or `count` would be more efficient on the database.  Alternatively do we need to check at all? Is it fine to return 0 `payments` when the `userId` provided does not exist?

Reply:
Yes, you are right, maybe I am to defensive so if something does not look right I rather throw an exception, and let the user give it another try.